src/libs/ascent/runtimes/flow_filters/ascent_runtime_relay_filters.cpp

```diff
@@ -578,13 +586,15 @@ verify_io_params(const conduit::Node &params,
 
 
 //-----------------------------------------------------------------------------
-void mesh_blueprint_save(const Node &data,
-                         const std::string &path,
-                         const std::string &file_protocol,
-                         int num_files,
-                         const Node &extra_opts,
-                         std::string &root_file_out)
+void
+mesh_blueprint_save(const Node &data,
+                    const std::string &path,
+                    const std::string &file_protocol,
+                    int num_files,
+                    const Node &extra_opts,
+                    std::string &root_file_out)
```

Comment:
I guess `root_file_out` wasn't used before, but was just a hook here for when we added the generate filename helper?

Now that it is getting set and used somewhere else in the case where we can generate the filename, do you handle the case where we cannot generate the filename? Does it still get used somewhere else? Does that make sense?

Reply:
I guess I don't see where `root_file_out` is plumbed to, but you should make sure that wherever it goes it is robust to the case where it does not exist.

Reply:
It was being used before but was always blank. The upshot was that the extracts path entry in the `ascent.info` output was always an empty string when saving meshes.

Reply:
do we want that behavior to remain? It might be more useful to add a message there saying that it couldn't generate the output filename b/c reasons. Or maybe it should be omitted in that case?

Reply:
I think an empty path is best for now, it's easier to check for that than to parse an error string.

Reply:
sounds good